packages/components/src/edit-in-place-control/index.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * External dependencies
+ */
+import { noop } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useState, useRef, useEffect } from '@wordpress/element';
+import { ENTER, ESCAPE } from '@wordpress/keycodes';
+
+/**
+ * Internal dependencies
+ */
+import Button from '../button';
+
+const baseInputCssClass = 'components-edit-in-place-control__input';
+const beforeTransitionInputCssClass = 'small';
+const afterTransitionInputCssClass = 'large';
+const withoutTransitionCssClass = `${ baseInputCssClass } ${ beforeTransitionInputCssClass }`;
+const withTransitionCssClass = `${ baseInputCssClass } ${ afterTransitionInputCssClass }`;
+const cancelEvent = ( event ) => (
+	event.preventDefault(), event.stopPropagation()
+);
+
+function EditInPlaceControl( { label = '', onClick = noop, onUpdate = noop } ) {
+	const [ edit, setEdit ] = useState( false );
+	const [ value, setValue ] = useState( label );
+	const [ inputCssClasses, setInputCssClasses ] = useState(
+		withoutTransitionCssClass
+	);
+
+	const prevValue = useRef();
+	const inputRef = useRef();
+	const buttonRef = useRef();
+
+	useEffect( () => {
+		prevValue.current = value;
+		if ( edit ) {
+			inputRef.current.focus();
+			inputRef.current.select();
+		} else {
+			buttonRef.current.focus();
+		}
+	}, [ edit ] );
+
+	return (
+		<>
+			{ edit ? (
+				<input
+					ref={ inputRef }
+					className={ inputCssClasses }
+					value={ value }
+					onChange={ ( event ) => {
+						setValue( event.target.value );
+					} }
+					onFocus={ () =>
+						setInputCssClasses( withTransitionCssClass )
+					}
+					onBlur={ () => {
+						setEdit( false );
+						onUpdate( value );
+						setInputCssClasses( withoutTransitionCssClass );
+					} }
+					onKeyDown={ ( event ) => {
+						if ( ENTER === event.keyCode ) {
```

Comment:
Hmm, yeah, I've noticed this before. It's used quite a lot in this codebase, which might due to the mousetrap dependency that's used for shortcuts. Could be worth a separate issue if there isn't one already.

Reply:
Edit in place control component has been moved to a separate PR https://github.com/WordPress/gutenberg/pull/28924
I have switched to using `key` there instead of deprecated `keyCode`